scene/gui/code_edit.cpp

```diff
@@ -1419,41 +1419,27 @@ void CodeEdit::fold_line(int p_line) {
 		/* End line is the same therefore we have a block of single line delimiters. */
 		if (end_line == p_line) {
 			for (int i = p_line + 1; i <= line_count; i++) {
-				if (i == line_count) {
-					end_line = line_count;
+				if ((in_string != -1 && is_in_string(i) == -1) || (in_comment != -1 && is_in_comment(i) == -1)) {
+					end_line = i - 1;
 					break;
 				}
 
-				if ((in_string != -1 && is_in_string(i) == -1) || (in_comment != -1 && is_in_comment(i) == -1)) {
-					end_line = i - 1;
+				if (i == line_count) {
+					end_line = line_count;
 					break;
 				}
 			}
 		}
 	} else {
 		int start_indent = get_indent_level(p_line);
 		for (int i = p_line + 1; i <= line_count; i++) {
-			if (get_line(p_line).strip_edges().size() == 0 || is_in_string(i) != -1 || is_in_comment(i) != -1) {
-				end_line = i;
-				continue;
-			}
-
-			if (i == line_count) {
-				/* Do not fold empty last line of script if any */
-				end_line = i;
-				if (get_line(i).strip_edges().size() == 0) {
-					end_line--;
-				}
-				break;
-			}
-
-			if ((get_indent_level(i) <= start_indent && get_line(i).strip_edges().size() != 0)) {
-				/* Keep an empty line unfolded if any */
-				end_line = i - 1;
-				if (get_line(i - 1).strip_edges().size() == 0 && i - 2 > p_line) {
-					end_line = i - 2;
+			if (get_line(i).strip_edges().size() != 0) {
+				if (get_indent_level(i) > start_indent) {
+					end_line = i;
+					continue;
+				} else if (is_in_string(i) == -1 && is_in_comment(i) == -1) {
```

Comment:
Nit: No need to `else if` can just `if` as we continue above.

Reply:
I actually don't like using `continue` and `break` as I find them less clear than if-elseif statements; it was in there originally and I forgot it was no longer needed. I'll remove that instead and leave the elseif, but tell me if you'd prefer your suggestion anyway (;

Reply:
That why it's `nit`, for me, `continue` and `break` are clearer :)

Reply:
Ah, ok. Now changed; I'll make sure to consider them more often when contributing